presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/DefaultCachingJdbcClientFactory.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.jdbc;
+
+import io.airlift.units.Duration;
+
+import javax.inject.Inject;
+
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Objects.requireNonNull;
+
+public class DefaultCachingJdbcClientFactory
+        implements CachingJdbcClientFactory
+{
+    private final JdbcClient jdbcClient;
+
+    @Inject
```

Comment:
Does it mean `CachingJdbcClient` is no longer `@Inject`-ed and https://github.com/prestosql/presto/blob/142c8e20fb14535064667f8a7978525aa6e1f6f5/presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/CachingJdbcClient.java#L62-L66 ctor should be removed?

or (likely), this is only for per-trnasaction cache?
then what about other cache?

Reply:
`CachingJdbcClient` is still created by Guava. There are two caches:
1. per-transaction cache
2. global cache (created by Guava) that is cross transactions.

This is similar mechanism to what we have in Hive